src/main/java/io/supertokens/webserver/api/oauth/OAuthClientsAPI.java

```diff
@@ -137,11 +139,46 @@ protected void doDelete(HttpServletRequest req, HttpServletResponse resp) throws
         }
     }
 
-    private JsonObject createJsonFromException(OAuthException exception){
+    @Override
+    protected void doPatch(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        JsonObject input = InputParser.parseJsonObjectOrThrowError(req);
+        InputParser.collectAllMissingRequiredFieldsOrThrowError(input, REQUIRED_INPUT_FIELDS_FOR_PATCH);
+
+        try {
+            AppIdentifier appIdentifier = getAppIdentifier(req);
+            Storage storage = enforcePublicTenantAndGetPublicTenantStorage(req);
+
+            JsonObject response = OAuth.updateOauthClient(super.main, appIdentifier, storage, input);
+            JsonObject postResponseBody = new JsonObject();
+            postResponseBody.addProperty("status", "OK");
+            postResponseBody.add("client", response);
+            sendJsonResponse(200, postResponseBody, resp);
+
+        } catch (OAuthAPIInvalidInputException exception) {
+            JsonObject errorResponse = createJsonFromException(exception, INVALID_INPUT_ERROR);
+            sendJsonResponse(400, errorResponse, resp);
+
+        } catch (OAuthClientUpdateException updateException) {
+            //for errors with the update from hydra, which are not invalid input errors
+            JsonObject errorResponse = createJsonFromException(updateException, OAUTH2_CLIENT_UPDATE_ERROR);
+            sendJsonResponse(200, errorResponse, resp);
```

Comment:
this should result in a 500 error from the core as well - cause i suspect hydra will only return 500 in case something like the db is down or something?

Reply:
At this endpoint hydra's behaviour doesn't seem to be consistent. It can also return a 500 for example if the input json is empty. (So, not just for only real internal server errors, but for like programmer errors and stuff too.. I couldn't get a guideline out of hydra about this error handling..)
I can change the return code to 500 in the core also, if you like, but I don't think that status code carries the message of what went wrong. (But of course, we can say that "hydra went wrong", which is an internal error - only it went wrong because a wrong input)

Reply:
We should send a 500, with a generic error - we should not expose to the user that we are using hydra

Reply:
Okay. The error hydra sends back in these cases is: `{error: "error", error_description: "The error is unrecognizable"}`. I would suggest to go with the more generic "Internal server error". Wdyt?

Reply:
Sure.